manticore/native/models.py

```diff
@@ -57,7 +59,37 @@ def _find_zero(cpu, constrs, ptr):
     return offset
 
 
-def strcmp(state, s1, s2):
+def _find_zeros(cpu, constrs, ptr):
+    """
+    Helper for finding the closest NULL or, effectively NULL byte from a starting address.
+
+    :param Cpu cpu:
+    :param ConstraintSet constrs: Constraints for current `State`
+    :param int ptr: Address to start searching for a zero from
+    :return: Offset from `ptr` to first byte that is 0 or an `Expression` that must be zero
+    """
+
+    offset = 0
+    zeros = []
+    while True:
+        byt = cpu.read_int(ptr + offset, 8)
+
+        if issymbolic(byt):
+            if Z3Solver.instance().can_be_true(constrs, byt == 0):
+                zeros.append(offset)
+            if not Z3Solver.instance().can_be_true(constrs, byt != 0):
+                break
```

Comment:
This is taking me a while to comprehend. Could you add an explanation to this, please

There are too many `can_be_true`, `not`, `==`, `!=` combinations and it's making my head spin, specifically the `not Z3Solver.instance().can_be_true(constrs, byt != 0)` expression 🙃

Reply:
Yeah I will add a comment to this. Everything I added also made my head spin and I needed a break when I was done lol, but the general idea is 

If the given byte can be 0:
   then add it to the list of zeros
If the byte is constrained to 0 (there is not a case where it can be anything but 0):
   break